Node script crashes when SOLR returns bad result

Node crashes on when BAD SOLR query is sent and SOLR returns error message

Here is traceback

`
web_1        | 151.207.250.51 - - [18/Jul/2016:16:24:48 +0000] "GET /bower_components/designpatterns/generated/styles/select2.png HTTP/1.1" 404 571 "http://bdr-sit.etc.uspto.gov/bower_components/designpatterns/generated/st
yles/vendor.css" "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.110 Safari/537.36" "-"
appserver_1  | http://10.201.19.197:8983/solr/oadata_3_shard1_replica3/select?q={!q.op=AND df=textdata}"judicial exception" OR "abstract idea" OR "natural phenomena" OR "natural phenomenon" OR "natural product" OR "law of
nature" OR “product of nature”&fq=type:oa&wt=json&indent=true&rows=20&start=0&hl=true&hl.snippets=10&hl.fl=textdata&hl.fragsize=200&hl.simple.pre=<code>&hl.simple.post=</code>&hl.usePhraseHighlighter=true&q.op=AND
appserver_1  | /server/boot/search.js:100
appserver_1  |                     took:body.responseHeader.QTime,
appserver_1  |                              ^
appserver_1  |
appserver_1  | TypeError: Cannot read property 'responseHeader' of undefined
appserver_1  |     at /server/boot/search.js:100:30
appserver_1  |     at Request.handleRequestResponse [as _callback] (/node_modules/unirest/index.js:461:25)
appserver_1  |     at Request.self.callback (/node_modules/unirest/node_modules/request/request.js:373:22)
appserver_1  |     at emitTwo (events.js:106:13)
appserver_1  |     at Request.emit (events.js:191:7)
appserver_1  |     at Request.<anonymous> (/node_modules/unirest/node_modules/request/request.js:1318:14)
appserver_1  |     at emitOne (events.js:101:20)
appserver_1  |     at Request.emit (events.js:188:7)
appserver_1  |     at IncomingMessage.<anonymous> (/node_modules/unirest/node_modules/request/request.js:1266:12)
appserver_1  |     at emitNone (events.js:91:20)
appserver_1  |     at IncomingMessage.emit (events.js:185:7)
web_1        | 2016/07/18 16:25:04 [error] 8#8: *203 upstream prematurely closed connection while reading response header from upstream, client: 151.207.250.51, server: localhost, request: "GET /newsearch?q=%22judicial+exc
eption%22+OR+%22abstract+idea%22+OR+%22natural+phenomena%22+OR+%22natural+phenomenon%22+OR+%22natural+product%22+OR+%22law+of+nature%22+OR+%E2%80%9Cproduct+of+nature%E2%80%9D&dataset=oa HTTP/1.1", upstream: "http://172.18.
0.3:3000/newsearch?q=%22judicial+exception%22+OR+%22abstract+idea%22+OR+%22natural+phenomena%22+OR+%22natural+phenomenon%22+OR+%22natural+product%22+OR+%22law+of+nature%22+OR+%E2%80%9Cproduct+of+nature%E2%80%9D&dataset=oa"
, host: "bdr-sit.etc.uspto.gov", referrer: "http://bdr-sit.etc.uspto.gov/newsearch?q=%2235+U.S.C.+112+%28b%29%22+OR+%2235+U.S.C.+112+2nd+paragraph%22+OR+%2235+U.S.C.+112+second+paragraph%22+OR+%2235+USC+112+%28b%29%22+OR+%
2235+USC+112+2nd+paragraph%22+OR+%2235+USC+112+second+paragraph%22&dataset=oa"
web_1        | 2016/07/18 16:25:04 [error] 8#8: *203 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 151.207.250.51, server: localhost, request: "GET /newsearch?q=%22judic
ial+exception%22+OR+%22abstract+idea%22+OR+%22natural+phenomena%22+OR+%22natural+phenomenon%22+OR+%22natural+product%22+OR+%22law+of+nature%22+OR+%E2%80%9Cproduct+of+nature%E2%80%9D&dataset=oa HTTP/1.1", upstream: "http://
172.18.0.3:3000/media/50x.html", host: "bdr-sit.etc.uspto.gov", referrer: "http://bdr-sit.etc.uspto.gov/newsearch?q=%2235+U.S.C.+112+%28b%29%22+OR+%2235+U.S.C.+112+2nd+paragraph%22+OR+%2235+U.S.C.+112+second+paragraph%22+O
R+%2235+USC+112+%28b%29%22+OR+%2235+USC+112+2nd+paragraph%22+OR+%2235+USC+112+second+paragraph%22&dataset=oa"
web_1        | 151.207.250.51 - - [18/Jul/2016:16:25:04 +0000] "GET /newsearch?q=%22judicial+exception%22+OR+%22abstract+idea%22+OR+%22natural+phenomena%22+OR+%22natural+phenomenon%22+OR+%22natural+product%22+OR+%22law+of+
nature%22+OR+%E2%80%9Cproduct+of+nature%E2%80%9D&dataset=oa HTTP/1.1" 502 575 "http://bdr-sit.etc.uspto.gov/newsearch?q=%2235+U.S.C.+112+%28b%29%22+OR+%2235+U.S.C.+112+2nd+paragraph%22+OR+%2235+U.S.C.+112+second+paragraph%
22+OR+%2235+USC+112+%28b%29%22+OR+%2235+USC+112+2nd+paragraph%22+OR+%2235+USC+112+second+paragraph%22&dataset=oa" "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.110 Safari/537.
36" "-"
appserver_1  |
appserver_1  | npm info lifecycle officeactions@1.0.1~start: Failed to exec start script
appserver_1  | npm ERR! Linux 4.4.11-23.53.amzn1.x86_64
appserver_1  | npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
appserver_1  | npm ERR! node v6.3.0
appserver_1  | npm ERR! npm  v3.10.3
appserver_1  | npm ERR! code ELIFECYCLE
appserver_1  | npm ERR! officeactions@1.0.1 start: `node .`
appserver_1  | npm ERR! Exit status 1
appserver_1  | npm ERR!
appserver_1  | npm ERR! Failed at the officeactions@1.0.1 start script 'node .'.
appserver_1  | npm ERR! Make sure you have the latest version of node.js and npm installed.
appserver_1  | npm ERR! If you do, this is most likely a problem with the officeactions package,
appserver_1  | npm ERR! not with npm itself.
appserver_1  | npm ERR! Tell the author that this fails on your system:
`

--- a/searchtool/server/boot/search.js
+++ b/searchtool/server/boot/search.js
@@ -50,7 +50,7 @@ exports.buildSearch = function (req, res) {
             var reqArray = req.query.documentcode;
             var reqArrayLen = req.query.documentcode.length;
             var reqLoopVar = reqArrayLen - 1;
-            var reqString = '';          
+            var reqString = '';
             for (var i = reqLoopVar; i >0; i--) {
                 reqString += reqArray[i] + "+OR+";
             }
@@ -109,7 +109,6 @@ exports.buildSearch = function (req, res) {
                 res.render('newview', {
                     total:0,
                     pagein:'',
-                    took:body.responseHeader.QTime,
                     term:q,
                     email: req.body.email,
                     accessOK: !!(! config.requireLogin || token.id)

